application/common/models/User.php

```diff
@@ -426,13 +435,42 @@ public function getDisplayName()
     }
 
     /**
-     * Is it time to nag the user to add an MFA?
-     *
-     * @return bool
+     * Based on current time and presence of MFA and Method options,
+     * determine which "nag" to present to the user.
      */
-    public function isTimeToNagForMfa()
+    public function getNagState()
     {
-        return strtotime($this->nag_for_mfa_after) < time();
+        /*
+         * Don't recalculate in case the date has changed since the last calculation.
+         */
+        if ($this->nagState !== null) return $this->nagState;
+
+        $now = time();
+
+        if (strtotime($this->nag_for_mfa_after) < $now) {
+            if (count($this->getVerifiedMfaOptions()) === 0) {
+                $this->nagState = self::NAG_ADD_MFA;
+            } else {
+                $this->nagState = self::NAG_REVIEW_MFA;
```

Comment:
So it looks like we'll be using the `nag_for_mfa_after` database field for two related but not quite identical purposes:
1) How long to wait before nagging the user to turn on MFA (by adding one), and
2) How long to wait (once they are using MFA) to have them review their MFA settings.

Is that right?

Those seem related enough that I don't see a problem with that (as long as the time-between-reviews can be different than the time-between-nags-to-add-mfa), but I wanted to make sure I understood it correctly and that we're doing that intentionally.

Reply:
Yes, that is correct. That's why I changed the `nag` json property to `add` in order to keep semantics clear. "nag" is a general term for user reminder, and can either be an "add" or "review"